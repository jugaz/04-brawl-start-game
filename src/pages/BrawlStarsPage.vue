<template>
    <h1 v-if="!brawl">Espere por favor...</h1>
    <div class="contenedor-brawlstartpage" v-else>   
        <h1>¿Quién es este brawl?</h1>

        <BrawlStarsPicutre :brawl-start-id="brawl.id" :show-brawl-start="showBrawlStart"/>
        <BrawlStarsOptions  
            :brawls="brawlStartArr"
            @selection-brawl-start="checkAnswer"/>

        <div class="response-brawlstartpage" v-if="showAnswer">
            <h2 class="fade-in">{{ message }}</h2>
            <button @click="newGame">
                NUEVO JUEGO
            </button>
        </div>
    </div>





</template>

<script>
import BrawlStarsPicutre from "@/components/BrawlStarsPicutre.vue"
import BrawlStarsOptions from "@/components/BrawlStarsOptions.vue"
import getBrawlStarsOptions from "@/helpers/getBrawlStartOptions"

export default {

    components: {
        BrawlStarsPicutre,
        BrawlStarsOptions
    },
    data() {
        return {
            brawlStartArr: [],
            brawl: null,
            showBrawlStart: false,
            showAnswer:false,
            message:'',
        }
    },

    methods: {
        async mixBrawlsArray() {
            this.brawlStartArr = await getBrawlStarsOptions()
            console.log(this.brawlStartArr)

            const rndInt = Math.floor( Math.random() * 4 )
            this.brawl = this.brawlStartArr[ rndInt ] 
        },
        
        checkAnswer( brawlId ) {

            this.showBrawlStart = true
            this.showAnswer = true
            console.log ("event",event.target)
            if( brawlId === this.brawl.id ) {
                this.message = `Excelente, era ${ this.brawl.name }`
                event.target.classList.add("success")
                document.querySelectorAll(".name-brawl").forEach(element => { 
                      element.classList.add("disabled")
                });
            }
            else {
                event.target.classList.add("fail")
                if(event.target.classList.contains("fail")) {
                    document.querySelectorAll(".name-brawl").forEach(element => {
                        
                        
                        if(this.brawl.name === element.innerHTML ) {
                             element.classList.add("success")
                        }
                        else {
                            element.classList.add("disabled")
                        }
                    });
                }
                this.message = `Oops, era ${ this.brawl.name }`
          
            }
        },
        newGame() {
            this.showBrawlStart = false
            this.showAnswer     = false
            this.brawlStartArr  = []
            this.brawl          = null
            this.mixBrawlsArray()
        }

    },
    mounted() {
        this.mixBrawlsArray()
    }
}
</script>

<style scoped>
    .response-brawlstartpage {
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        margin-bottom: 30px;
        padding-inline-start: 40px;
    }
    .response-brawlstartpage  button {
        background-color: #fbbf24;
        border-color: #343a40;
        border-radius: .25rem;
        border: 1px solid transparent;
        color: #fff;
        cursor: pointer;
        font-size: 1rem;
        font-weight: 400;
        line-height: 1.5;
        margin-bottom: 10px;
        padding: 5px 0;
        text-align: center;
        transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
        user-select: none;
        vertical-align: middle;
        white-space: nowrap;
        width: 250px;
    }

    .response-brawlstartpage button:hover {
        background-color: #fcd34d;
    }
</style>