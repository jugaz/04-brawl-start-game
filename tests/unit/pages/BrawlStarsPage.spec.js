import { shallowMount, mount } from "@vue/test-utils"
import BrawlStarsPage from "@/pages/BrawlStarsPage"

import { mockBrawlstars } from "../mocks/brawlstarts.mock";


describe('BrawlStarsPage component', () => {

    let wrapper

    beforeEach(() => {
        wrapper = shallowMount( BrawlStarsPage )
        
    })

    test('debe de hacer match con el snapshot', () => {
        expect( wrapper.html() ).toMatchSnapshot()    
    
    })

    test('debe de llamar mixBrawlsArray al montar', () => {

        const mixBrawlsArraySpy = jest.spyOn( BrawlStarsPage.methods, 'mixBrawlsArray' )
        const wrapper = shallowMount( BrawlStarsPage )

        expect( mixBrawlsArraySpy ).toHaveBeenCalled() 
    })

    test('debe de hacer match con el snapshot cuando cargan los brawls',() => {
    
        const wrapper = shallowMount( BrawlStarsPage, {

            data() {
                return {
                    brawlStartArr: mockBrawlstars,
                    brawl: mockBrawlstars[0],
                    showBrawlStart: false,
                    showAnswer:false,
                    message:'',
                }
            },
        
        })

        expect( wrapper.html() ).toMatchSnapshot()  
    
    })
    
    test('debe de mostrar los componentes de BrawlStarsPicutre y BrawlStarsOptions',() => {
    


          const wrapper = shallowMount( BrawlStarsPage, {

            data() {
                return {
                    brawlStartArr: mockBrawlstars,
                    brawl: mockBrawlstars[0],
                    showBrawlStart: false,
                    showAnswer:false,
                    message:'',
                }
            },
        })

        const picture = wrapper.find('brawl-stars-picutre-stub')
        const options = wrapper.find('brawl-stars-options-stub')

        expect(picture.exists()).toBeTruthy()
        expect(options.exists()).toBeTruthy()

        expect(picture.attributes('brawlstartid')).toBe("1")
        expect(options.attributes('brawls')).toBeTruthy()
    })

    test('pruebas con checkAnswer ', async () => {
        const wrapper = shallowMount( BrawlStarsPage, {

            data() {
                return {
                    brawlStartArr: mockBrawlstars,
                    brawl: mockBrawlstars[0],
                    showBrawlStart: false,
                    showAnswer:false,
                    message:'',
                }
            },
        })

    
        await wrapper.vm.checkAnswer(1)
        
        expect( wrapper.find('h2').exists() ).toBeTruthy()
        expect( wrapper.vm.showBrawlStart ).toBeTruthy()
        expect( wrapper.find('h2').text() ).toBe(`Excelente, era ${ mockBrawlstars[0].name }`)
    
        await wrapper.vm.checkAnswer(10)
        expect( wrapper.vm.message ).toBe(`Oops, era ${  mockBrawlstars[0].name }`)
    })
    
    
    

})