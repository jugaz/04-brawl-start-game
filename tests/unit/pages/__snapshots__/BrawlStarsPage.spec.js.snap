// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrawlStarsPage component debe de hacer match con el snapshot 1`] = `<h1>Espere por favor...</h1>`;

exports[`BrawlStarsPage component debe de hacer match con el snapshot cuando cargan los brawls 1`] = `
<div class="contenedor-brawlstartpage">
  <h1>¿Quién es este brawl?</h1>
  <brawl-stars-picutre-stub brawlstartid="1" showbrawlstart="false"></brawl-stars-picutre-stub>
  <brawl-stars-options-stub brawls="[object Object],[object Object],[object Object],[object Object]"></brawl-stars-options-stub>
  <!--v-if-->
</div>
`;
